### ISSUE TITLE
Maui media picker on Windows Platform error

### ISSUE THREAD

**Issue by TajinderSingh-1** [dotnet/maui#7660]
I have created a very basic MauiApp because I wanted to try the MediaPicker on the Windows Platform.

Thus I copied the code from the official [documentation][1] and tried to run my application

  [1]: https://docs.microsoft.com/en-us/dotnet/maui/platform-integration/device-media/picker?tabs=windows

However if I add `<uap:Capability Name="webcam"/>` to the `Package.appxmanifest` file as suggested in the documentaion, and run the application it gives me the following error:

    
    Error		DEP0700: Registration of the app failed. [0x80080204] error 0xC00CE169: App 
    manifest validation error: The app manifest must be valid as per schema: Line 39, Column 
    21, Reason: 'webcam' violates enumeration constraint of 'documentsLibrary 
    picturesLibrary videosLibrary musicLibrary enterpriseAuthentication 
    sharedUserCertificates userAccountInformation removableStorage appointments contacts 
    phoneCall blockedChatMessages objects3D voipCall chat'.
    The attribute 'Name' with value 'webcam' failed to parse.	MauiApp3			
			

So in order to solve this problem I tried to change the capability from `<uap:Capability Name="webcam"/>` to `<DeviceCapability Name="webcam"/>`.

In this way I can run the application without errors, but `photo` is always null:

    public async void TakePhoto(object sender, EventArgs e)
    {
        if (MediaPicker.Default.IsCaptureSupported)
        {
            FileResult photo = await MediaPicker.Default.CapturePhotoAsync();
            
            if (photo != null)
            {
                // save the file into local storage
                string localFilePath = Path.Combine(FileSystem.CacheDirectory, photo.FileName);

                using Stream sourceStream = await photo.OpenReadAsync();
                using FileStream localFileStream = File.OpenWrite(localFilePath);

                await sourceStream.CopyToAsync(localFileStream);
            }
            else
            {
                // *** IT ALWAYS ENTERS IN THE ELSE CLAUSE ***
                // *** BECAUSE photo IS ALWAYS NULL ***
                CounterBtn.Text = $"Capture is supported but {photo} is null";
            }
        }
    }

Note: The function above is called when I click to this button that I've defined in `MainPage.xaml` file:

            <Button 
                x:Name="ImageBtn"
                Text="Take Photo"
                SemanticProperties.Hint="Take Image"
                Clicked="TakePhoto"
                HorizontalOptions="Center" />

Stackoverflow question: https://stackoverflow.com/questions/72450077/maui-media-picker-on-windows-platform-error

**Comment by janseris** [comment]
https://github.com/dotnet/maui/issues/7465
https://github.com/dotnet/maui/issues/6660

**Comment by richardrigutins** [comment]
Actually, with a slight modification in the custom CameraCaptureUI class, @GiampaoloGabba's workaround can be used to capture both photos and videos on Windows:
```csharp
public async Task<StorageFile> CaptureFileAsync(CameraCaptureUIMode mode)
{
	var extension = mode == CameraCaptureUIMode.Photo ? ".jpg" : ".mp4";

	var currentAppData = ApplicationData.Current;
	var tempLocation = currentAppData.LocalCacheFolder;
	var tempFileName = $"CCapture{extension}";
	var tempFile = await tempLocation.CreateFileAsync(tempFileName, CreationCollisionOption.GenerateUniqueName);
	var token = Windows.ApplicationModel.DataTransfer.SharedStorageAccessManager.AddFile(tempFile);

	var set = new ValueSet();
	if (mode == CameraCaptureUIMode.Photo)
	{
		set.Add("MediaType", "photo");
		set.Add("PhotoFileToken", token);
	}
	else
	{
		set.Add("MediaType", "video");
		set.Add("VideoFileToken", token);
	}

	var uri = new Uri("microsoft.windows.camera.picker:");
	var result = await Windows.System.Launcher.LaunchUriForResultsAsync(uri, _launcherOptions, set);
	if (result.Status == LaunchUriStatus.Success && result.Result != null)
	{
		return tempFile;
	}

	return null;
}
```

**Comment by GiampaoloGabba** [comment]
@richardrigutins thank you!
When i tried the workaround for videos, i forgot to set the "**VideoFileToken**"  `ValueSet` (i kept using PhotoFileToken) and the app crashed when trying to take videos 😄 

I confirm that now videocapturing is owrking

**Comment by TajinderSingh-1** [comment]
To me it is showing the type or namespace Windows  could not be found error

**Comment by richardrigutins** [comment]
> To me it is showing the type or namespace Windows could not be found error

It's a platform-specific implementation, you need to add it to the _Platforms > Windows_ folder for example (see the [documentation](https://docs.microsoft.com/en-us/dotnet/maui/platform-integration/invoke-platform-code)).

Here you can find the code for a sample app that uses this workaround: [repository](https://github.com/richardrigutins/maui-sample-windows-camera-workaround).

**Comment by Shidesu** [comment]
Is there any update on that issue WinUI or MAUI side?

**Comment by janseris** [comment]
@jsuarezruiz Hi is anything happening with this issue? It's been around for more than 6 months.

**Comment by mattleibow** [comment]
Hi folks, just browsing through the older issues and saw this one with some good looking ways to make things work. Not sure if this is all your code or you got it from somewhere else? I want to add some credits here: https://github.com/dotnet/maui/pull/13220

If this is your code, you can go ahead and make a PR as well.

I am having some issues, not sure if you saw these before - on the PR so we don't clog up this issue with dev fixes: #13220
